include/xtensor/xcontainer.hpp

```diff
@@ -715,67 +607,86 @@ namespace xt
+    template <class D, layout L>
+    inline xstrided_container<D, L>::xstrided_container(inner_shape_type&& shape, inner_strides_type&& strides) noexcept
         : base_type(), m_shape(std::move(shape)), m_strides(std::move(strides))
     {
         m_backstrides = make_sequence<inner_backstrides_type>(m_shape.size(), 0);
         adapt_strides(m_shape, m_strides, m_backstrides);
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::shape_impl() noexcept -> inner_shape_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::shape_impl() noexcept -> inner_shape_type&
     {
         return m_shape;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::shape_impl() const noexcept -> const inner_shape_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::shape_impl() const noexcept -> const inner_shape_type&
     {
         return m_shape;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::strides_impl() noexcept -> inner_strides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::strides_impl() noexcept -> inner_strides_type&
     {
         return m_strides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::strides_impl() const noexcept -> const inner_strides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::strides_impl() const noexcept -> const inner_strides_type&
     {
         return m_strides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::backstrides_impl() noexcept -> inner_backstrides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::backstrides_impl() noexcept -> inner_backstrides_type&
     {
         return m_backstrides;
     }
 
-    template <class D>
-    inline auto xstrided_container<D>::backstrides_impl() const noexcept -> const inner_backstrides_type&
+    template <class D, layout L>
+    inline auto xstrided_container<D, L>::backstrides_impl() const noexcept -> const inner_backstrides_type&
     {
         return m_backstrides;
     }
 
     /**
+     * Return the layout of the container
+     * @return layout of the container
+     */
+    template <class D, layout L>
+    xt::layout xstrided_container<D, L>::layout()
+    {
+        return m_layout;
+    }
+
+    /**
      * Reshapes the container.
      * @param shape the new shape
+     * @param force force reshaping, even if the shape stays the same (default: false)
      */
-    template <class D>
-    inline void xstrided_container<D>::reshape(const shape_type& shape)
+    template <class D, layout L>
+    inline void xstrided_container<D, L>::reshape(const shape_type& shape, bool force)
     {
-        if (shape != m_shape)
+        if (m_layout == xt::layout::dynamic)
         {
-            reshape(shape, layout::row_major);
+            m_layout = xt::layout::row_major;  // fall back to row major
```

Comment:
I think this should be moved in the if statement below, where the reshape is really done. Else you may end with a layout inconsistent with the strides.